spec/models/incognito_spec.rb

```diff
@@ -0,0 +1,71 @@
+require 'spec_helper'
+
+describe Mumuki::Domain::Incognito do
+  let(:user) { Mumuki::Domain::Incognito }
+  let(:organization) { create(:organization) }
+
+  it 'can mock an AR relation' do
+    Assignment.new submitter: user
+  end
+
+  describe 'ensure_enabled!' do
+    it { expect { user.ensure_enabled! }.to_not raise_error }
+  end
+
+  describe 'assignment fooling' do
+    let(:assignment) { user.build_assignment(exercise, organization) }
+
+    context 'when exercise has no default content' do
+      let(:exercise) { create(:exercise) }
+
+      it { expect(assignment).to be_an Assignment }
+      it { expect(assignment.submitter).to be user }
+      it { expect(assignment.current_content).to eq '' }
+    end
+
+    context 'when exercise has default content' do
+      let(:exercise) { create(:exercise, default_content: '...') }
+
+      it { expect(assignment.current_content).to eq '...' }
+    end
+  end
+
+  describe 'guide fooling' do
+    let(:guide) { create(:guide, exercises: [ create(:exercise), create(:exercise) ]) }
+
+    describe 'completion_percentage_for', organization_workspace: :test do
+      it { expect(guide.completion_percentage_for(user)).to eq 0 }
+    end
+
+    describe 'next_exercise', organization_workspace: :test do
+      # TODO this looks weird. Why do we need a non-polymorphic
```

Comment:
Currently we aren't using indicators for this one as progress is shared between organizations.
This change was made and had to be rolled back because of that.